src/ersilia_pack/parsers/yaml_install_parser.py

```diff
@@ -0,0 +1,89 @@
+import os
+import re
+import yaml
+from .install_parser import InstallParser
+
+class YAMLInstallParser(InstallParser):
+    def __init__(self, file_dir, conda_env_name=None):
+        self.file_type = "install.yml"
+        self.file_name = os.path.join(file_dir, self.file_type)
+        
+        # Load YAML data before call
+        self.data = self._load_yaml()
+        
+        # Call the super class constructor
+        super().__init__(self.file_name, conda_env_name)
+
+    def _load_yaml(self):
+        with open(self.file_name, 'r') as file:
+            return yaml.safe_load(file)
+
+    def _get_python_version(self):
+        if "python" not in self.data or not isinstance(self.data["python"], str):
+            raise ValueError("Python version must be a string")
+        return self.data["python"]
+
+    def _get_commands(self):
+        if "commands" not in self.data:
+            raise KeyError("Missing 'commands' key in YAML file")
+        
+        commands = []
+        for command in self.data["commands"]:
+            if isinstance(command, dict) and command.get("tool") == "pip":
+                commands.append(self._process_pip_command(command))
+            else:
+                # Handle other command types or formats if needed
+                commands.append(command)
+        return commands
+
+    def _process_pip_command(self, command):
+        """
+        Reconstruct pip install command from YAML entry.
+
+        Example input YAML format for pip commands:
+        - tool: pip
+          package: torchvision
+          version: 0.17.1
+          flags:
+            --index-url: https://download.pytorch.org/whl/cpu
+          url: git+https://github.com/pytorch/vision.git
+
+        Output: 
+        - "pip install torchvision==0.17.1 --index-url https://download.pytorch.org/whl/cpu"
+        - "pip install git+https://github.com/pytorch/vision.git"
+        """

```

Comment:
@OlawumiSalaam Can you share an example YAML? I don't think we need additional keys in the YAML. We still want to keep only `commands`, and then have list entries for that. So something like, 

```yaml
commands:
- ["pip", "torchvision", "0.17.1", "--index-url", "<url>"]
```

Asking a user to specify a yaml this detailed would not make for a good experience. 

A